src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
+import { getTodos } from './api';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [option, setOption] = useState('All');
+  const [inputValue, setInputValue] = useState('');
+  const [selectEye, setSelectEye] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(todos => setTodosFromServer(todos))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = todosFromServer.filter(todo => {
+    const titleOfMatches = todo.title
+      .toLowerCase()
+      .includes(inputValue.toLowerCase());
+    const statusSearch = (() => {
+      switch (option) {
+        case 'active':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return true;
+      }
+    })();
+
+    return titleOfMatches && statusSearch;
+  });
```

Comment:
The `filteredTodos` function combines filtering by title and status in a single operation. While this works, consider separating these concerns or optimizing the logic for better readability and potential performance improvements. For instance, you could first filter by status and then by title, or vice versa, depending on which is more likely to reduce the dataset size earlier.